packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -783,8 +795,10 @@ class HotRunner extends ResidentRunner {
     String reason,
     void Function(String message) onSlow,
   }) async {
+    final Map<FlutterDevice, List<FlutterView>> viewCache = <FlutterDevice, List<FlutterView>>{};
```

Comment:
In the case of a hot reload the views must be the same before and after so caching this is safe for the lifetime of this method (approximately...)